src/Tribe/Admin/Move_Tickets.php

```diff
@@ -0,0 +1,640 @@
+			}
+
+			$this->ticket_provider = $provider;
+		}
+	}
+
+	/**
+	 * Given an attendee array, attempts to determine which ticket provider
+	 * owns it (Woo, RSVP, etc).
+	 *
+	 * @param array $attendee
+	 *
+	 * @return string|null
+	 */
+	protected function get_ticket_provider( array $attendee ) {
+		if ( ! isset( $attendee[ 'product_id' ] ) ) {
+			return null;
+		}
+
+		$ticket_type = Tribe__Tickets__Tickets::load_ticket_object( $attendee[ 'product_id' ] );
+
+		if ( ! $ticket_type ) {
+			return null;
+		}
+
+		if ( property_exists( $ticket_type, 'provider_class' ) ) {
+			return $ticket_type->provider_class;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Adds a "Move" option to the attendee screen bulk action selector.
+	 *
+	 * There is not a corresponding bulk action handler, as such, because when this
+	 * is selected further handling will be managed via JS (and interaction will be
+	 * through a modal interface).
+	 *
+	 * @param array $actions
+	 *
+	 * @return array
+	 */
+	public function bulk_actions( array $actions ) {
+		$actions[ 'move' ] = _x( 'Move', 'attendee screen bulk actions', 'event-tickets' );
+		return $actions;
+	}
+
+	/**
+	 * Responds to ajax requests for a list of supported post types.
+	 */
+	public function get_post_types() {
+		if ( ! wp_verify_nonce( $_POST['check' ], 'move_tickets' ) ) {
+			wp_send_json_error();
+		}
+
+		wp_send_json_success( array( 'posts' => $this->get_post_types_list() ) );
+		exit();
+	}
+
+	/**
+	 * Returns a list of post types for which tickets are currently enabled.
+	 *
+	 * The list is expressed as an array in the following format:
+	 *
+	 *     [ 'slug' => 'name', ... ]
+	 *
+	 * @return array
+	 */
+	protected function get_post_types_list() {
+		$types_list = array( 'all' => __( 'All supported types', 'tribe-tickets' ) );
+
+		foreach ( Tribe__Tickets__Main::instance()->post_types() as $type ) {
+			$pto = get_post_type_object( $type );
+			$types_list[ $type ] = $pto->label;
+		}
+
+		return $types_list;
+	}
+
+	/**
+	 * Responds to requests for a list of possible destination posts.
+	 */
+	public function get_post_choices() {
+		if ( ! wp_verify_nonce( $_POST['check' ], 'move_tickets' ) ) {
+			wp_send_json_error();
+		}
+
+		$args = wp_parse_args( $_POST, array(
+			'post_type'    => '',
+			'search_terms' => '',
+			'ignore'       => '',
+		) );
+
+		wp_send_json_success( array( 'posts' =>  $this->get_possible_matches( $args ) ) );
+		exit();
+	}
+
+	/**
+	 * Returns a list of posts that could be possible homes for a ticket
+	 * type, given the constraints in optional array $request (if not set,
+	 * looks in $_POST for the corresponding values):
+	 *
+	 * - 'post_type': string or array of post types
+	 * - 'search_term': string used for searching posts to narrow the field
+	 *
+	 * @param array|null $request post parameters (or looks at $_POST if not set)
+	 *
+	 * @return array
+	 */
+	protected function get_possible_matches( array $request = null ) {
```

Comment:
This method is identical to the one declared [here](https://github.com/moderntribe/event-tickets/pull/229/files#diff-41c69c826caa255771804f564897194fR91).